packages/eth2.0-types/test/unit/validator.test.ts

```diff
@@ -0,0 +1,100 @@
+import {describe, it} from "mocha";
+import {Validator} from "../../src/types/misc";
+import {getRegistry, Registry} from "../../src/types/validator";
+import BN from "bn.js";
+import {expect} from "chai";
+import { Gwei } from "../../src/types/primitive";
+
+describe("ValidatorRegistry", () => {
+  let validators: Partial<Validator>[] = [];
+  let validatorRegistry: Registry<Partial<Validator>, Gwei>;
+  const EFFECTIVE_BALANCES: BN[] = [];
+  const TOTAL_ITEM = 6;
+
+  before(() => {
+    for (let i = 0; i < TOTAL_ITEM; i++) {
+      EFFECTIVE_BALANCES.push(new BN(i));
+    }
+  });
+
+  beforeEach(() => {
+    validators = [];
+    for (let i = 0; i < TOTAL_ITEM; i++) {
+      validators.push({effectiveBalance: EFFECTIVE_BALANCES[i]});
+    }
+    validatorRegistry = getRegistry<Partial<Validator>, Gwei>(validators, (val: Partial<Validator>) => val.effectiveBalance);
```

Comment:
if you overide array constructor you could so validatorRegistry = new ValidatorRegistry(validators)

Reply:
Here we have 2 params. Do you mean `validatorRegistry = new ValidatorRegistry(validators, (val: Partial<Validator>) => val.effectiveBalance`?

Reply:
whats the purpose of second function?

Reply:
that's to define the field you want to index.
For validator, I defined a convenient one which already mean we want to index pubkey: `export const getValidatorRegistry = (validators: Validator[]): ValidatorRegistry => {`.
For this test, I want to define that I want to index `effectiveBalance` field inside validator.

Reply:
ValidatorRegistry is only indexed by publicKey, we can create more genric one later if thwre is a need

Reply:
to get ValidatorRegistry, I already implemented the shortcut `getValidatorRegistry(validators: Validator[])` without having to define the 2nd argument so it's already convenient.

If we find a way to support overriding array constructor, I'll drop the generic code to support ValidatorRegistry by public key only. Otherwise if we go with this approach, having the generic code does no harm.